backend/src/v5/schemas/tickets/templates.js

```diff
@@ -23,16 +23,38 @@ const {
 	presetModulesProperties,
 	propTypes } = require('./templates.constants');
 
+const { isArray, isString } = require('../../utils/helper/typeCheck');
 const { types, utils: { stripWhen } } = require('../../utils/helper/yup');
 const Yup = require('yup');
 const { cloneDeep } = require('../../utils/helper/objects');
-const { isString } = require('../../utils/helper/typeCheck');
 const { propTypesToValidator } = require('./validators');
 
+const TemplateSchema = {};
+
 const defaultFalse = stripWhen(Yup.boolean().default(false), (v) => !v);
+const nameSchema = types.strings.title.min(1);
+
+const pinColSchema = Yup.lazy((val) => {
+	if (val === undefined) return Yup.mixed().strip();
+	if (isArray(val)) return types.color3Arr;
+
+	return Yup.object({
+		property: Yup.object({
+			name: nameSchema.required(),
+			module: nameSchema,
+		}),
+
+		mapping: Yup.array().of(Yup.object({
+			default: types.color3Arr,
+			value: Yup.mixed().when('default', (def, schema) => (def ? schema.strip() : schema.required())),
+			color: types.color3Arr.when('default', (def, schema) => (def ? schema.strip() : schema.required())),
+		})).test('Color mapping', 'Must contain default entry', (arr) => arr.filter((obj) => !!obj.default).length === 1),
```

Comment:
you could change the message to Must contain one default entry to cover the case the user tries to enter two